### ISSUE TITLE
[com.google.fonts/check/name/license_url] https://openfontlicense.org is not recognized as the OFL’s URL

### ISSUE THREAD

**Issue by dscorbett** [fonttools/fontbakery#4418]
## Observed behavior

`com.google.fonts/check/name/license_url` warns if a font uses the OFL’s old URL but skips if the font is modified to use the OFL’s new URL.

With the old URL:
```
     WARN OFL url is no longer "https://scripts.sil.org/OFL".                   
          Use 'https://openfontlicense.org' instead. [code: deprecated-ofl-url] 
```

With the new URL:
```
     SKIP Could not infer the font license. Please ensure                       
          NameID 13 (LICENSE DESCRIPTION) is properly set.                      
```

## Expected behavior

`com.google.fonts/check/name/license_url` passes if a font uses the OFL’s new URL.

## Resources and steps needed to reproduce

```zsh
pip install fontbakery==0.10.9
curl -O https://www.ponomar.net/files/Vilnius.zip
unzip Vilnius.zip
ttx -t name Vilnius-Regular.otf
sed -i.bak 's|https://scripts.sil.org/OFL|https://openfontlicense.org|' Vilnius-Regular.ttx
ttx -m Vilnius-Regular.otf Vilnius-Regular.ttx
fontbakery check-googlefonts -c com.google.fonts/check/name/license_url Vilnius-Regular.otf
fontbakery check-googlefonts -c com.google.fonts/check/name/license_url -l SKIP Vilnius-Regular#1.otf
```

**Comment by felipesanches** [comment]
Today I've been working on this one. I think the entire check needs to be reimplemented. The check should be smarter to detect the intended license (by having both name IDs 13 and 14 as well as any license file as source of truth). And only after detecting the most likely user intent, then reporting WARNs or FAILs that are meaningful to the user.

The check should also have tests, which is currently lacks.

I'm working on both aspects and will soon send a PR.

**Comment by vv-monsalve** [comment]
> And only after detecting the most likely user intent, then reporting WARNs or FAILs that are meaningful to the user.

I'm having trouble understanding what you mean by "most likely user intent" in regard to this check's smartness.

**Comment by simoncozens** [comment]
This must be fixed before #4072.

**Comment by vv-monsalve** [comment]
> The check should be smarter to detect the intended license (by having both name IDs 13 and 14 as well as any license file as source of truth)

Fontbakery should detect license URL consistency in those three sources and report a Fail if there is a mismatch. This is being ignored. [See an example in this PR](https://github.com/google/fonts/pull/7458#discussion_r1541780686)

**Comment by simoncozens** [comment]
@felipesanches This is a symptom of a more general problem. When we think of some new thing which needs checking, our response is normally to add another check for it. This means we have a large number of small, discrete checks, which individually are easy to understand and reason about but which don't necessarily "communicate" with one another. I would prefer to move towards a smaller number of checks which tell you everything about a particular topic - i.e. license URLS in all the different places they can appear - because then they can ensure consistency together.

**Comment by dscorbett** [comment]
This was fixed in version 0.11.0 by #4437.
